builder/azure/arm/template_factory.go

```diff
@@ -56,7 +56,41 @@ func GetKeyVaultDeployment(config *Config, secretValue string) (*resources.Deplo
 	return createDeploymentParameters(*doc, params)
 }
 
+func GetSpecializedVirtualMachineDeployment(config *Config) (*resources.Deployment, error) {
```

Comment:
I wonder if we can use some config attribute to determine that we want a specialised VM deployment, and invoke that when relevant in `GetVirtualMachineDeployment`?

Reply:
@lbajolet-hashicorp so the problem there is we don't pass in the config to GetVirtualMachineDeployment, and changing it was a bit of a messier change as its an interface used all over the place.

Reply:
Fair enough, if this is a messy change we can leave it as-is I would think.